luda-editor/luda-editor-client/src/app/editor/clip_editor/camera_clip_editor/image_browser/types.rs

```diff
@@ -1,68 +1,145 @@
+            directory.url.clone()
+        } else {
+            format!("{}/", directory.url)
+        };
+        self.url.starts_with(&directory_url_with_suffix_slash)
+    }
+    pub(crate) fn is_just_under_directory(&self, directory: &ImageBrowserDirectory) -> bool {
+        let path = Path::new(&self.url);
+        let parent = path.parent().unwrap();
+        let parent_path = parent.to_str().unwrap().to_string();
+        parent_path == directory.url
+    }
+    pub fn get_url(&self) -> String {
+        self.url.clone()
     }
 }
 
-impl ImageBrowserItem {
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ImageBrowserDirectory {
+    url: String,
+}
+impl ImageBrowserDirectory {
+    pub fn new(url: String) -> ImageBrowserDirectory {
+        Self { url }
+    }
+    pub fn root() -> Self {
+        Self {
+            url: "/".to_string(),
+        }
+    }
+    pub(crate) fn navigate_to_parent(&mut self) {
+        let path = Path::new(&self.url);
+        let parent = path.parent().unwrap();
+        let parent_path = parent.to_str().unwrap().to_string();
+        self.url = parent_path;
+    }
     pub(crate) fn to_string(&self) -> String {
-        match self {
-            ImageBrowserItem::Back => vec!["Back".to_string()],
-            ImageBrowserItem::Character(character) => vec![character.clone()],
-            ImageBrowserItem::CharacterPose(character, pose) => {
-                vec![character.clone(), pose.clone()]
-            }
-            ImageBrowserItem::CharacterPoseEmotion(character, pose, emotion) => {
-                vec![character.clone(), pose.clone(), emotion.clone()]
-            }
-            ImageBrowserItem::Empty => {
-                vec!["Empty".to_string()]
+        self.url.clone()
+    }
+    pub(crate) fn is_root(&self) -> bool {
+        self.url == "/"
+    }
+    pub(crate) fn top(&self) -> String {
+        let path = Path::new(&self.url);
+        path.file_name().unwrap().to_str().unwrap().to_string()
+    }
+    pub(crate) fn is_just_under_directory(&self, directory: &ImageBrowserDirectory) -> bool {
+        let path = Path::new(&self.url);
+        let parent = path.parent().unwrap();
+        let parent_path = parent.to_str().unwrap().to_string();
+        parent_path == directory.url
+    }
+    pub(crate) fn is_recursively_under_directory(&self, directory: &ImageBrowserDirectory) -> bool {
+        self.url.starts_with(format!("{}/", directory.url).as_str())
+    }
+    pub(crate) fn get_diff(&self, directory: &ImageBrowserDirectory) -> (String, String) {
+        if self.url == directory.url {
+            return ("".to_string(), "".to_string());
+        }
+        let self_chunks = self.url.split("/").collect::<Vec<&str>>();
+        let directory_chunks = directory.url.split("/").collect::<Vec<&str>>();
+
+        /*
+        expected result
+          self: /a/b
+          directory: /a/bc
+          --> (b, c)
+        */
```

Comment:
Is it typo?

This is what I expect
```
       expected result
          self: /a/b
          directory: /a/bc
          --> (b, bc)
```

Reply:
oh my gosh, yes it's typo.

Reply:
That's because the AI didn't suggest good code to me...